File: src/Tree/129. 求根到叶子节点数字之和/sumNumbers.go
package main

import (
	"math"
)

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

var sum int

func sumNumbers(root *TreeNode) int {
	path := make([]int, 0)
	travel(root, path)
	return sum
}
func travel(node *TreeNode, path []int) {
	if node.Left == nil && node.Right == nil {
		path = append(path, node.Val)
		l := len(path)
		j := l - 1
		for i := 0; i < l; i++ {
			sum += path[i] * int(math.Pow10(j))
			j--
		}
	} else {
		path = append(path, node.Val)
		if node.Left != nil {
			travel(node.Left, path)
		}
		if node.Right != nil {
			travel(node.Right, path)
		}
	}
}

File: src/Tree/129. 求根到叶子节点数字之和/sumNumbers.php
<?php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($value) { $this->val = $value; }
 * }
 */
class TreeNode {
    public $val = null;
    public $left = null;
    public $right = null;
    function __construct($value) { $this->val = $value; }
}
class Solution {

    /**
     * @param TreeNode $root
     * @return Integer
     */
    public $sum=0;
    function sumNumbers($root) {
        $this->travel($root,[]);
        return  $this->sum;
    }

    function travel($node,$path){
        if($node->left===null && $node->right===null){
            $path[] = $node->val;
            $len = count($path);
            $j=$len-1;
            for($i=0;$i<$len;$i++){
                $val= $path[$i]*pow(10,$j);
                $this->sum+=$val;
                $j--;
            }
        }else{
            $path[] = $node->val;
            if($node->left !== null){
                $this->travel($node->left,$path);
            }
            if($node->right !== null){
                $this->travel($node->right,$path);
            }
        }
    }
}